releases/release-5.0.3.md

```diff
@@ -0,0 +1,123 @@
+---
+title: TiDB 5.0.3 Release Notes
+---
+
+# TiDB 5.0.3 Release Notes
+
+Release date: June 30, 2021
+
+TiDB version: 5.0.3
+
+## Compatibility Changes
+
+## New Features
+
++ TiFlash
+
+    - Support cast string as double push down [#2068](https://github.com/pingcap/tics/pull/2068)
+    - Support pushing down the `str_to_date` function to TiFlash [#2063](https://github.com/pingcap/tics/pull/2063)
+
+## Improvements
+
++ TiDB
+
+    - Support planner pushing down the TopN operator to MPP. [#25162](https://github.com/pingcap/tidb/pull/25162)
+
++ TiFlash
+
+    - Use multi threads for not joined data in right outer join. [#2094](https://github.com/pingcap/tics/pull/2094)
+
+## Bug Fixes
+
++ TiDB
+
+    - Fix the issue that an incorrect result is returned when using merge join on the `SET` types. [#25694](https://github.com/pingcap/tidb/pull/25694)
+    - Fix the `IN` expression arguments corruption issue. [#25666](https://github.com/pingcap/tidb/pull/25666)
+    - Avoid sessions of GC being affected by global variables [#25609](https://github.com/pingcap/tidb/pull/25609)
+    - Make sure limit outputs no more columns than its child [#25517](https://github.com/pingcap/tidb/pull/25517)
+    - Planner: fix a panic caused by sinking a Limit with inlined Proj into IndexLookUp when accessing a partition table [#25139](https://github.com/pingcap/tidb/pull/25139)
+    - Executor: fix ifnull bug when arg is enum/set [#25116](https://github.com/pingcap/tidb/pull/25116)
+    - Generate correct number of rows when all agg funcs are pruned [#25062](https://github.com/pingcap/tidb/pull/25062)
+    - Executor: make the ParallelApply be safe to be called again after returning empty results [#25011](https://github.com/pingcap/tidb/pull/25011)
+    - Fix the case that there could be duplicate ranges for multi-column index. [#24635](https://github.com/pingcap/tidb/pull/24635)
+    - Fix a nulleq bug [#24633](https://github.com/pingcap/tidb/pull/24633)
+    - Executor: fix data race of parallel apply operator [#24345](https://github.com/pingcap/tidb/pull/24345)
+    - Fix an issue that sorting on index-merge results in partition union reports 'index out of range'. [#24155](https://github.com/pingcap/tidb/pull/24155)
+
++ TiFlash
+
+    - Fix the issue that TiFlash nodes keep restart because of split failure. [#2219](https://github.com/pingcap/tics/pull/2219)
+    - Fix the bug that TiFlash can not GC delta data under rare case [#2185](https://github.com/pingcap/tics/pull/2185)
+    - Fix the bug that tiflash wrongly add padding for non-binary chars in cast function [#2092](https://github.com/pingcap/tics/pull/2092)
+    - Fix a bug that aggregation query with complex group by column may got wrong results. [#2072](https://github.com/pingcap/tics/pull/2072)
+    - Fix the bug that TiFlash may panic under heavy write pressure [#2055](https://github.com/pingcap/tics/pull/2055)
```

Comment:
+ TiFlash

    - Fix the issue that TiFlash keeps restarting because of split failure
    - Fix the potential issue that TiFlash can not GC delta data
    - Fix the bug that TiFlash adds wrong padding for non-binary chars in `CAST` function
    - Fix the issue of incorrect results when handling aggregation queries with complex `GROUP BY` columns
    - Fix the TiFlash panic issue that occurs under heavy write pressure
    - Fix the panic that occurs when right join if the right jon key is not nullalbe and the left join key is nullable
    - Fix the potential issue that read-index requests cost long time